flink-tests/src/test/java/org/apache/flink/test/checkpointing/RescalingITCase.java

```diff
@@ -135,14 +146,16 @@ public void setup() throws Exception {
 
             Configuration config = new Configuration();
 
-            final File checkpointDir = temporaryFolder.newFolder();
+            checkpointDir = temporaryFolder.newFolder();
             final File savepointDir = temporaryFolder.newFolder();
 
             config.setString(StateBackendOptions.STATE_BACKEND, currentBackend);
             config.setString(
                     CheckpointingOptions.CHECKPOINTS_DIRECTORY, checkpointDir.toURI().toString());
             config.setString(
                     CheckpointingOptions.SAVEPOINT_DIRECTORY, savepointDir.toURI().toString());
+            config.setBoolean(CheckpointingOptions.INCREMENTAL_CHECKPOINTS, incrementalCheckpoint);
+            config.setInteger(CheckpointingOptions.MAX_RETAINED_CHECKPOINTS, 20);
```

Comment:
I think the large retained checkpoints here is to avoid the detected checkpoint not be deleted on next successful checkpoint for next rescale. However, this is not totally safe unless you configure this value to a extremly large number.

I think we can refer to the logic of `ResumeCheckpointManuallyITCase` to add another `RescaleCheckpointManuallyITCase`.